WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -13,4 +13,38 @@ def competition_message_for_user(competition, user)
                 end
     messages.join(' ')
   end
+
+  def announced_class(days_announced)
+    if days_announced <= 21
+      "alert-danger"
+    elsif days_announced <= 28
+      "alert-orange"
+    else
+      "alert-green"
+    end
+  end
+
+  def report_class(days_report)
+    if days_report > 21
+      "alert-danger"
+    elsif days_report > 14
+      "alert-orange"
+    elsif days_report > 7
+      "alert-success"
+    else
+      "alert-green"
+    end
+  end
+
+  def results_class(days_results)
```

Comment:
`report_class` and `results_class` look identical. Should they be the same function?

A fun (but maybe obscure) patterns for these things is

    def self.report_class(days_report)
      level = [7, 14, 21].select{|d| days_report >  d}.count
      ["alert-green", "alert-success", "alert-orange", "alert-danger"][level]
    end

_(I have no other comments. This just caught my interest)_